doc/lib_input_terminal.md

```diff
@@ -0,0 +1,81 @@
+// This file is distributed under the terms of the MIT license, (c) the KSLib team
+
+## lib_input_terminal.
+
+`lib_input_terminal` provides functions for getting user entered strings or numbers though terminal inputs.
+
+### terminal_input_string
+
+args:
+  1) col - (scalar), The column to start the print of the current string at as required by `PRINT x AT(col,row).`
+  2) row - (scalar), The row to start the print of the current string at as required by `PRINT x AT(col,row).`
+  3) maxLength - (scalar), The maximum length of the string that can be returned,
+    defaulted to the terminal width minus the col argument.
+  4) inValue - (string or scalar), The string or scalar to pre-populate the input space with,
+    defaulted to an empty string.
+  5) cursorBlink - (boolean), Determines if the character denoting where the next char entered by the user will go will cycle between `█` and `_` or stay a stedy `_`
+    defaulted to true.
+  
+returns:
+  A string
+  
+description:
+  This function takes key presses on the kOS terminal and adds them into a single string until the user presses enter to end return the string.
+  While the string is being added to it will be printed on the terminal at the location set by the arguments.
+  The control keys that are recognized by this function are backspace, delete, and enter/return.
+    Pressing enter/return ends the loop and returns the current string.
+    Pressing backspace removes the last charter from the current string until the string is empty.
+    Pressing delete sets the current string to an empty string.
+	
+### terminal_input_number
+
+args:
+  1) col - (scalar), The column to start the print of the current string at as required by `PRINT x AT(col,row).`
+  2) row - (scalar), The row to start the print of the current string at as required by `PRINT x AT(col,row).`
+  3) maxLength - (scalar), The one greater than the maximum length of the string that can be returned,
+    defaulted to the terminal width minus the col argument.
+  4) inValue - (string or scalar), The string or scalar to pre-populate the input space with,
+    defaulted to an empty string.
+    if not a valid number will be discarded and replaced with the default.
+  5) cursorBlink - (boolean), Determines if the character denoting where the next char entered by the user will go will cycle between "█" and "_" or stay a steady "_"
+    defaulted to true.
+  
+returns:
+  A string
+
+description:
+  This function takes a limited set of key presses on the kOS terminal and adds them into a single string.  As this function is intended for the input of scalars it only adds to a limited set of key presses, the keys it will add to the string are "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", and "." with "+" and "-" as additional control characters.
```

Comment:
```suggestion
  This function takes key presses on the kOS terminal and adds them into a single string for displaying a number.  As such this function only works with a limited set of keys. The keys it will add to the string are "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", and "." with "+" and "-" for denoting sign.
```